library_information_validator.py

```diff
@@ -0,0 +1,495 @@
+            index_search = charRE.search(index)
+            if(bool(index_search) and index != "noIndex"):
+                logger.error('The index sequence \"{}\" in pool {} contains invalid characters.'\
+                ' Allowed characters: A/T/C/G/N/a/t/c/g/n/-'
+                .format(index, pool_name))
+                quit()
+
+            # check that indexes within a pool have minimum diversity
+            for i in range(index_count,len(index_seq)):
+                if lev.distance(index.lower(), index_seq[i].lower()) < LibrarySheet.MAX_DISTANCE:
+                    logger.warning('The index sequences {} and {} in pool {}'\
+                    ' display low diversity (only {} nt difference).'\
+                    .format(index,index_seq[i], pool_name, lev.distance(index.lower(), index_seq[i].lower())))
+                    warning_low_div += 1
+            index_count += 1
+
+        # checks index length
+        warning_index_length = 0
+        warning_index_balance = 0
+        if(not sindex):
+            index_length = []
+            for index in index_seq:
+                index_length.append(len(index))
+            count_length = Counter(index_length)
+
+            if(len(count_length) > 1):
+                logger.warning('There are {} different index lengths in pool {}, please double check the sequences.'\
+                .format(len(count_length),pool_name))
+                warning_index_length += 1
+
+            # checks color balance in the pool
+            max_length = sorted(count_length.keys())[-1]
+            min_length = sorted(count_length.keys())[0]
+            index_list_colour = []
+            for index in index_seq:
+                index_colour = index.replace('T','G').replace('A','R').replace('C', 'R')
+                index_list_colour.append(list(index_colour))
+
+            for row_nr in range(0,max_length):
+                column = []
+                for row in index_list_colour:
+                    try:
+                        column.append(row[row_nr])
+                    except IndexError:
+                        pass
+                count_colour = Counter(column)
+                if(len(count_colour)<2 and sum(count_colour.values()) > 1):
+                    logger.warning('Indexes in pool {} unbalanced at position {}'\
+                    .format(pool_name, row_nr+1))
+                    warning_index_balance += 1
+
+        # returns warnings for check summary
+        return(warning_low_div, warning_index_length, warning_index_balance)
+
+
+def main(input_sheet, config_statusDB):
+    # Instantiate the LibrarySheet object
+    sheetOI = LibrarySheet(input_sheet)
+    # get Project Information from couchDB
+    Project_Information, project_plate_ID = sheetOI.ProjectInfo(config_statusDB)
+    # validate the project name to ensure correct identification in couchDB
+    sheetOI.validate_project_Name(Project_Information, project_plate_ID)
```

Comment:
If the `sheetOI`  class is generating these variables, I think the `ProjectInfo` function should be able to save the values to `self. Project_Information` and `self. project_plate_ID`. Then you won't have to pass them to the function like this.